libraries/AP_Scripting/examples/param_get_set_test.lua

```diff
@@ -1,43 +1,29 @@
 -- This script is a test of param set and get
 local count = 0
 
+-- for fast param acess it is better to get a param object,
+-- this saves the code searching for the param by name every time
+local VM_I_Count = Parameter()
+assert(VM_I_Count:init('SCR_VM_I_COUNT'),'get SCR_VM_I_COUNT failed')
+
 function update() -- this is the loop which periodically runs
 
   -- get and print all the scripting parameters
-  local value = param:get('SCR_ENABLE')
-  if value then
-    gcs:send_text(6, string.format('LUA: SCR_ENABLE: %i',value))
-  else
-    gcs:send_text(6, 'LUA: get SCR_ENABLE failed')
-  end
-  value = param:get('SCR_VM_I_COUNT')
-  if value then
-    gcs:send_text(6, string.format('LUA: SCR_VM_I_COUNT: %i',value))
-  else
-    gcs:send_text(6, 'LUA: get SCR_VM_I_COUNT failed')
-  end
-  value = param:get('SCR_HEAP_SIZE')
-  if value then
-    gcs:send_text(6, string.format('LUA: SCR_HEAP_SIZE: %i',value))
-  else
-    gcs:send_text(6, 'LUA: get SCR_HEAP_SIZE failed')
-  end
-  value = param:get('SCR_DEBUG_LVL')
-  if value then
-    gcs:send_text(6, string.format('LUA: SCR_DEBUG_LVL: %i',value))
-  else
-    gcs:send_text(6, 'LUA: get SCR_DEBUG_LVL failed')
-  end
+  local value = assert(param:get('SCR_ENABLE'),'get SCR_ENABLE failed')
+  gcs:send_text(6, string.format('LUA: SCR_ENABLE: %i', value))
 
-  -- increment the script heap size by one
-  local heap_size = param:get('SCR_HEAP_SIZE')
-  if heap_size then
-    if not(param:set('SCR_HEAP_SIZE',heap_size + 1)) then
-      gcs:send_text(6, 'LUA: param set failed')
-    end
-  else
-    gcs:send_text(6, 'LUA: get SCR_HEAP_SIZE failed')
-  end
+  value = assert(param:get('SCR_DEBUG_LVL'),'get SCR_DEBUG_LVL failed')
+  gcs:send_text(6, string.format('LUA: SCR_DEBUG_LVL: %i', value))
+
+  value = assert(param:get('SCR_HEAP_SIZE'),'get SCR_HEAP_SIZE failed')
+  gcs:send_text(6, string.format('LUA: SCR_HEAP_SIZE: %i', value))
+
+  -- increment the script heap size by one using 'by name' method
+  assert(param:set('SCR_HEAP_SIZE',value + 1),'param set failed')
+
+  -- increment the VM I count by one using direct accsess method
+  gcs:send_text(6, string.format('LUA: SCR_VM_I_COUNT: %i', VM_I_Count:get()))
+  assert(VM_I_Count:set( VM_I_Count:get() + 1),'param set failed')
```

Comment:
The test here should actually explicitly try and manipulate parameters that don't exist to document what should happen.

Reply:
removed assets and replaced with messages, also try's to init a parm that does not exist